component/src/main/java/org/wso2/siddhi/runner/beam/SiddhiDoFnOperator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.runner.beam;
+
+import org.apache.beam.runners.core.DoFnRunner;
+import org.apache.beam.runners.core.SideInputReader;
+import org.apache.beam.runners.core.SimpleDoFnRunner;
+import org.apache.beam.runners.core.StateInternals;
+import org.apache.beam.runners.core.StepContext;
+import org.apache.beam.runners.core.TimerInternals;
+import org.apache.beam.runners.core.construction.ParDoTranslation;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.PInput;
+import org.apache.beam.sdk.values.POutput;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.WindowingStrategy;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Creates and manages {@link SimpleDoFnRunner} given the {@link PTransform} and {@link PCollection}
+ * in order to perform {@link ParDo} transform.
+ * @param <InputT>
+ * @param <OutputT>
+ * @param <TransformT>
+ */
+public class SiddhiDoFnOperator
+        <InputT extends PInput, OutputT extends POutput, TransformT extends PTransform<InputT, OutputT>> {
+
+    public final AppliedPTransform<InputT, OutputT, TransformT> transform;
+    public DoFnRunner<InputT, OutputT> delegate;
+    public PCollection<InputT> collection;
+    public PipelineOptions options;
+    public SideInputReader sideInputReader;
+    public EventChunkOutputManager outputManager;
+    public TupleTag mainOutputTag;
+    public List<TupleTag<?>> additionalOutputTags;
+    public StepContext stepContext;
+    public Coder<InputT> inputCoder;
+    public Map<TupleTag<?>, Coder<?>> outputCoders;
+    public WindowingStrategy windowingStrategy;
+    public DoFn<InputT, OutputT> fn;
```

Comment:
can we give more meaningful variable name ?

Reply:
These are actually the same variable names used in a different Beam runner implementation(Flink runner). Therefore I've adopted the names here too.

Reply:
okay